src/Sensor/SensorController.java

```diff
@@ -1,55 +1,102 @@
+    private final static String TIMESTAMP_KEY = "timestamp";
+
+    private final static String DATE_PATTERN = "yyyy-MM-dd'T'HH:mm:ss";
+
+    private JSONObject dataForMonitor = new JSONObject();
+    private JSONArray measuresJsonArray = new JSONArray();
+    private String resourceID = "";
+    private String hostname = "";
+    private String hostDescription = "";
+
+
+    private UDPConnectionController udpController = new UDPConnectionController();
     private List<Measurement> measurementsList = new ArrayList<>();
-    private String data;
 
-    public void startSensor(){
-        if(registerSensor()) {
-            chooseMeasurements();
-            sendMetadata();
+    private Sigar sigar;
 
-            //noinspection InfiniteLoopStatement
-            while(true) run();
+    public void startSensor(){
+        configureSensor();
 
+        //noinspection InfiniteLoopStatement
+        while(true) {
+            sendMetadata();
+            try {
+                Thread.sleep(1000);
+            } catch(InterruptedException ex) {
+                Thread.currentThread().interrupt();
+            }
+            for(int i=0; i<10; ++i)
+                run();
         }
-        else{
-            System.out.println("Rejestracja nieudana, program konczy dzialanie.");
-        }
+
     }
 
-    private boolean registerSensor(){
-        //this function will register sensor and return true on succes
-        return true;
+
+
+    private void configureSensor(){
+        getHostInformationFromUser();
+        getMonitorAdressFromUser();
+
+        sigar = new Sigar();
+        resourceID = UUID.randomUUID().toString();
+
+        chooseMeasurements();
+    }
+    
+    private void getMonitorAdressFromUser(){
+        Scanner reader = new Scanner(System.in);
+        System.out.println("Enter monitor ip adress ");
+        udpController.setIPAddress(reader.nextLine());
+        System.out.println("Enter port number ");
+        udpController.setPortNumber(reader.nextLine());
+    }
+    private void getHostInformationFromUser(){
+        Scanner reader = new Scanner(System.in);
+        System.out.println("Enter hostname ");
+        hostname = reader.nextLine();
+        System.out.println("Enter host description ");
+        hostDescription = reader.nextLine();
     }
 
     private void chooseMeasurements(){
-        //this function will allows to choose proper measurements
-        //in reality it will probably just add all avaible measurements
-        Sigar sigar = new Sigar();
         measurementsList.add(new MemoryMeasurement(sigar));
         measurementsList.add(new CpuMeasurement(sigar));
     }
 
-    private void sendMetadata(){
-        //this function will send metadata
-    }
-
     private void run(){
-        prepateMeasurementsData();
-        sendMeasurementsData();
+        prepareMeasurementsData();
+        udpController.sendData(dataForMonitor.toJSONString());
```

Comment:
Isn't sendData bool function? What about false result, we simply ignore unsuccessful data send?